pyaedt/modules/Material.py

```diff
@@ -1992,16 +1993,248 @@ def set_magnetic_coercivity(self, value=0, x=1, y=0, z=0):
         return self.update()
 
     @pyaedt_function_handler()
-    def set_electrical_steel_coreloss(self, kh=0, kc=0, ke=0, kdc=0, cut_depth=0.0001):
-        """Set Electrical Steel Type Core Loss.
+    def get_core_loss_coefficients(
+        self,
+        points_list_at_freq,
+        core_loss_model_type="Electrical Steel",
+        thickness="0.5mm",
+        conductivity=0,
+        coefficient_setup="w_per_cubic_meter",
+    ):
+        """Compute core loss coefficients from the loss characteristics (B-P Curve) at a given frequency.
+
+        Parameters
+        ----------
+        points_list_at_freq : dict
+            points_list_at_freq is a dictionary of BP values for a given frequency.
+            Key is the frequency and values are the list of points (BP curve).
+        core_loss_model_type : str, optional
+            Core loss model type. It can be ``Electrical Steel`` or ``Power Ferrite``.
+            The default value is ``Electrical Steel``.
+        thickness : str, optional
+            Thickness provided as value + unit.
+            The default value is ``0.5mm``.
+        conductivity : float, optional
+            Material conductivity.
+            The default value is ``0``.
+        coefficient_setup : str, optional
+            Core loss unit.
+            Possible values are: ``w_per_cubic_meter``, ``kw_per_cubic_meter``, ``w_per_lb``, ``w_per_kg``.
+            The default value is ``w_per_cubic_meter``.
+
+        Returns
+        -------
+        list
+            List of core loss coefficients.
+            Returns Kh, Kc, Ke coefficients if core loss model is ``Electrical Steel``.
+            Returns Cm, X, Y if core loss model is ``Power Ferrite``.
+
+        Examples
+        --------
+        This example shows how to get core loss coefficients for Electrical Steel core loss model.
+
+        >>> from pyaedt import Maxwell3d
+        >>> m3d = Maxwell3d()
+        >>> box = m3d.modeler.create_box([-10, -10, 0], [20, 20, 20], "box_to_split")
+        >>> box.material = "magnesium"
+        >>> coefficients = m3d.materials["magnesium"].get_core_loss_coefficients(
+        ...                                                        points_list_at_freq={60 : [[0, 0], [1, 3], [2, 7]]},
+        ...                                                        thickness="0.5mm",
+        ...                                                        conductivity=0)
+        >>> print(coefficients)
+        >>> m3d.release_desktop(True, True)
+        """
+        if not isinstance(points_list_at_freq, dict):
+            self.logger.error("Points list at frequency must be provided as a dictionary.")
+            return False
```

Comment:
Here you should raise a TypeError
```
raise TypeError("Points list at frequency must be provided as a dictionary.")
```
The pyaedt_function_handler will catch the Exception and return False.

Reply:
@Alberto-DM is it also logged if we throw the TypeError?

Reply:
Excellent idea.
You can write in the log at first and then throw the TypeError.